haystack/document_stores/pinecone.py

```diff
@@ -459,69 +690,163 @@ def get_documents_by_id(
-        if not self.pinecone_indexes.get(index, False):
-            raise ValueError(f"No index named {index} found in Pinecone.")
+        index = str(index or self.index)
+        index = _sanitize_index_name(index)
+        if index not in self.pinecone_indexes:
+            raise DocumentStoreError(
+                f"Index named '{index}' does not exist. Try reinitializing PineconeDocumentStore() and running "
+                f"'update_embeddings()' to create and populate an index."
+            )
 
         stats = self.pinecone_indexes[index].describe_index_stats()
-        # if no namespace return zero
-        count = stats["namespaces"][""]["vector_count"] if "" in stats["namespaces"] else 0
+        # if no embeddings namespace return zero
+        count = (
+            stats["namespaces"][self.embedding_namespace]["vector_count"]
+            if self.embedding_namespace in stats["namespaces"]
+            else 0
+        )
         return count
 
-    def update_document_meta(self, id: str, meta: Dict[str, str], index: str = None):
+    def update_document_meta(self, id: str, meta: Dict[str, str], namespace: str = None, index: str = None):
         """
-        Update the metadata dictionary of a document by specifying its string id
+        Update the metadata dictionary of a document by specifying its string id.
+
+        :param id: ID of document to update.
+        :param meta: Dictionary of new metadata.
+        :param namespace: Optional namespace to update documents from. If not specified, defaults to the embedding
+            namespace (vectors) if it exists, otherwise the document namespace (no-vectors).
+        :param index: Optional index name for where to update documents from.
         """
-        index = index or self.index
-        index = self._sanitize_index_name(index)
-        if index in self.pinecone_indexes:
-            doc = self.get_documents_by_id(ids=[id], index=index, return_embedding=True)[0]
-            if doc.embedding is not None:
-                self.pinecone_indexes[index].upsert(vectors=([id], [doc.embedding.tolist()], [meta]))
 
-        super().update_document_meta(id=id, meta=meta, index=index)
+        index = str(index or self.index)
+        index = _sanitize_index_name(index)
+        if index not in self.pinecone_indexes:
+            raise DocumentStoreError(
+                f"Index named '{index}' does not exist. Try reinitializing PineconeDocumentStore() and running "
+                f"'update_embeddings()' to create and populate an index."
+            )
+
+        if namespace is None:
+            if self.get_embedding_count() > 0:
+                namespace = self.embedding_namespace
+            else:
+                namespace = self.document_namespace
+
+        doc = self.get_documents_by_id(ids=[id], index=index, return_embedding=True)[0]
+        if doc.embedding is not None:
+            meta = {**meta, **{"content": doc.content}}
```

Comment:
:scroll: Redundant braces
```suggestion
            meta = {"content": doc.content, **meta}
```